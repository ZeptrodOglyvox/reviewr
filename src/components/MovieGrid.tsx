import React, { FunctionComponent, useState } from "react";
import MovieIcon, { type MovieIconProps } from "./MovieIcon";
import "../styles/MovieGrid.css";

interface OwnProps {
    movies: any[]; // FIXME: Update with the model type
}

type MovieGridProps = OwnProps;

const MovieGrid: FunctionComponent<MovieGridProps> = ({ movies }) => {
    const [query, setQuery] = useState("");

    const filteredMovieIcon = movies
        .filter((m: MovieIconProps) =>
            m.title.match(new RegExp(query.trim(), "i")),
        )
        .map(MovieIcon);

    return (
        <>
            <div className="heading">
                <h2>Top Movies</h2>
                <form>
                    <label htmlFor="search">Search</label>
                    <input
                        id="search"
                        name="search"
                        type="text"
                        value={query}
                        onChange={(e) => {
                            e.preventDefault();
                            setQuery(e.target.value);
                        }}
                    />
                </form>
            </div>
            <div className="movie-icon-grid">{filteredMovieIcon}</div>
        </>
    );
};

export default MovieGrid;
