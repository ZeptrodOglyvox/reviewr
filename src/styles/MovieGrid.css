.movie-icon-grid {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(auto-fit, 200px);
    grid-auto-rows: 250px;
    align-items: center;
    justify-items: center;
}