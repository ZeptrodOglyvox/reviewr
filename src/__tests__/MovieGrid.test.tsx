import React from "react";
import { act, render, screen } from "@testing-library/react";
import MovieGrid from "../components/MovieGrid";
import "@testing-library/jest-dom";

test("empty list", async () => {
    render(<MovieGrid movies={[]} />);
    expect(await screen.findByTestId("empty-message-id")).toHaveTextContent(
        "Ok people, nothing to see here, move along...",
    );
});

test("happy path", async () => {
    const { asFragment } = render(
        <MovieGrid
            movies={[
                {
                    _id: "1",
                    title: "movie 1",
                    image: "http://img.com",
                },
                {
                    _id: "2",
                    title: "movie 2",
                    image: "http://img.com",
                },
            ]}
        />,
    );

    await act(async () => {
        expect(await asFragment()).toMatchSnapshot();
    });
});
