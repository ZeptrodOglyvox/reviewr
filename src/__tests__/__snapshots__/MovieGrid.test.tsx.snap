// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy path 1`] = `
<DocumentFragment>
  <div
    class="heading"
  >
    <h2>
      Top Movies (2)
    </h2>
    <form>
      <label
        for="search"
      >
        Search
      </label>
      <input
        id="search"
        name="search"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    class="movie-icon-grid"
    data-testid="movie-icon-grid"
  >
    <div
      class="movie-icon"
    >
      <a
        class="icon-overlay"
        href="#"
      >
        <p>
          movie 1
        </p>
      </a>
      <img
        src="http://img.com"
      />
    </div>
    <div
      class="movie-icon"
    >
      <a
        class="icon-overlay"
        href="#"
      >
        <p>
          movie 2
        </p>
      </a>
      <img
        src="http://img.com"
      />
    </div>
  </div>
</DocumentFragment>
`;
